packages/customize-widgets/src/components/inspector/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState, createPortal } from '@wordpress/element';
+import { BlockInspector } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+export { default as BlockInspectorButton } from './block-inspector-button';
+
+const container = document.createElement( 'div' );
+
+function InspectorSectionMeta( { collapse } ) {
+	return (
+		<div className="customize-section-description-container section-meta">
+			<div className="customize-section-title">
+				<button // eslint-disable-line react/forbid-elements
+					className="customize-section-back"
+					tabIndex="0"
+					onClick={ ( event ) => {
+						event.preventDefault();
+						collapse();
+					} }
+				>
+					<span className="screen-reader-text">Back</span>
+				</button>
+				<h3>
+					<span className="customize-action">
+						{ __( 'Customizing ▸ Widgets' ) }
+					</span>
+					{ __( 'Block Settings' ) }
+				</h3>
+			</div>
+		</div>
+	);
+}
+
+export default function Inspector( { isExpanded, collapse } ) {
+	const [ busy, setBusy ] = useState( false );
+
+	useEffect( () => {
+		const parent = document.getElementById( 'customize-theme-controls' );
+
+		parent.appendChild( container );
+
+		return () => {
+			parent.removeChild( container );
+		};
+	}, [] );
+
+	useEffect( () => {
+		setBusy( true );
+
+		const openedSidebarSection = document.querySelector(
+			'.control-section-sidebar.open'
+		);
+
+		if ( isExpanded ) {
+			openedSidebarSection.classList.add( 'is-inspector-open' );
+		} else {
+			openedSidebarSection.classList.remove( 'is-inspector-open' );
+		}
+
+		const timer = setTimeout( () => {
```

Comment:
Maybe should probably do both and race them? In case for some reasons the event does not fire. I think I've run into that before but forgot the root cause 🤔, maybe I'm overthinking though.

Reply:
I'd prefer if we can avoid hardcoded numbers but do what you gotta do 😅

Reply:
If only we are using CSS-in-JS 😆